common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -278,24 +278,11 @@ public ICacheRecord save(@NonNull AccountRecord accountRecord,
             );
         } else {
             // Save to the processUid cache... or create a new one
-            MsalOAuth2TokenCache targetCache = getTokenCacheForClient(
-                    idTokenRecord.getClientId(),
-                    idTokenRecord.getEnvironment(),
```

Comment:
I see.  We know !isFoci, so the checks for the foci-nature inside of getTokenCacheForClient were a complete waste of time.

Reply:
You got it

Reply:
It's interesting that in this case initializeProcessUidCache was unconditionally called... was there any way we would ever execute the below else statement?  OK, only in the case where the metadata lookup was null.